server.js

```diff
@@ -3388,10 +3394,16 @@ cache(function(data, match, sendBadge, request) {
 
         label = tag !== 'latest' ?  tag : '';
         if (asset_name !== 'total') {
-          label += ' ' + '[' + asset_name + ']';
+          if (label) {
+            label += ' ';
+          }
+          label += '[' + asset_name + ']';
         }
       }
-      badgeData.text[1] = metric(downloads) + ' ' + label;
+      badgeData.text[1] = metric(downloads);
+      if (label) {
+        badgeData.text[1] += ' ' + label;
+      }
```

Comment:
This code could be made clearer using [`Array.prototype.join`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/join):

```js
const labelWords = [metric(downloads)];
if (tag !== latest) {
  labelWords.push(tag);
}
if (asset_name !== 'total') {
  labelWords.push(`[${asset_name}]`);
}
badgeData.text[1] = labelWords.join(' ');
```

This might avoid future bugs too!

Reply:
Great idea! Thanks.